`VRisc` service implementation

uselessgoddess opened the pull request.

uselessgoddess committed:
Add `VRisc` service project files

uselessgoddess committed:
Bump thin implementation

uselessgoddess committed:
Add simple thin clean architecture dummy implementation

uselessgoddess committed:
Basic implementation of service without any emulation payload (only dummy emulator)

uselessgoddess committed:
Add ffi bindings to native implementation of risc-v emulator:
- fix dram import as base64
- add cpu/bus dto's

uselessgoddess committed:
Basic implementation of `VRisc` service:
- use clean architecture
- add ffi bindings
- proof of emulation concept

uselessgoddess committed:
Add docker support

MefistoSamael commented on VRisc/Api/Controllers/StatesController.cs:
remove all checks from controllers

MefistoSamael commented on VRisc/Api/Controllers/StatesController.cs:
use use cases. All logic move to them. Same for other controller

MefistoSamael commented on VRisc/UseCases/Emulation/EmulationStatesService.cs:
move to infrastructure

MefistoSamael commented on VRisc/Core/Interfaces/IEmulationStatesService.cs:
move to application

MefistoSamael commented on VRisc/UseCases/Emulation/EmulationTaskManager.cs:
move to infrastructure

MefistoSamael commented on VRisc/Core/Interfaces/IEmulationTaskManager.cs:
move to application

uselessgoddess committed:
Fix dependency injection:
- move out logic from controllers into handlers
- fix `Program.cs` configuration

uselessgoddess committed:
Merge branch 'feat/vrisc-service' into feat/vrisc-service-docker

uselessgoddess committed:
Merge pull request #4 from uselessgoddess/feat/vrisc-service-docker

Add docker support to `VRisc` service

uselessgoddess committed:
Merge branch 'dev' into feat/vrisc-service

uselessgoddess committed:
Merge branch 'dev' into feat/vrisc-service

uselessgoddess committed:
Fix solution after services merging

uselessgoddess committed:
fix docker to build new project directories

uselessgoddess committed:
Merge pull request #6 from uselessgoddess/feat/vrisc-service-fix

fix docker to build new project directories

uselessgoddess committed:
Add grpc client to `VRisc` service

MefistoSamael commented on VRisc/UseCases/Handlers/TasksHandler.cs:
separate each method into different file

MefistoSamael replied:
same for states handler

MefistoSamael commented on VRisc/Api/VRisc.Api.csproj:
you broke dependency rule

MefistoSamael commented on VRisc/Core/Interfaces/IEmulationStatesService.cs:
move to application

MefistoSamael commented on VRisc/Core/Interfaces/IEmulationTaskManager.cs:
move to applicatoin

uselessgoddess committed:
Fix dependencies order

uselessgoddess committed:
Merge remote-tracking branch 'origin/feat/vrisc-service' into feat/vrisc-service

uselessgoddess committed:
Merge branch 'feat/compiler-service-hangfire' into feat/(vrisc,compiler)-rabbitmq

# Conflicts:
#	docker-compose.yaml

uselessgoddess committed:
Add `RabbitMQ` as message broker:
- add ability to compile check

uselessgoddess committed:
Merge branch 'feat/vrisc-service-grpc' into feat/(vrisc,compiler)-rabbitmq

# Conflicts:
#	VRisc/Infrastructure/DependencyInjection.cs
#	VRisc/Infrastructure/VRisc.Infrastructure.csproj
#	VRisc/UseCases/Handlers/StatesHandler.cs

uselessgoddess committed:
Full rabbitmq implementation:
- bump grpc client into `VRisc`
- add protos for rabbit communication
- move hub to use cases
- fix rabbit connection issues

uselessgoddess committed:
Merge branch 'feat/vrisc-service' into feat/vrisc-service-grpc

uselessgoddess committed:
Merge pull request #7 from uselessgoddess/feat/vrisc-service-grpc

Add grpc client to `VRisc` service

uselessgoddess committed:
Merge branch 'feat/vrisc-service' into feat/(vrisc,compiler)-rabbitmq

uselessgoddess committed:
Merge pull request #10 from uselessgoddess/feat/(vrisc,compiler)-rabbitmq

Add RabbitMQ to `VRisc` and `Compiler`

uselessgoddess committed:
After-merge fixes

uselessgoddess committed:
Use CQRS with MediatR

uselessgoddess committed:
Add ocelot as api gateway

uselessgoddess committed:
Add tests template:
- add `cargo test` to `Dockerfile`

uselessgoddess committed:
Add tests:
- fix `Compiler` imports
- add tests to `zxc` compiler:
  - add `cargo test` to `Dockerfile`

uselessgoddess committed:
Fix missing `identity` service at api gateway

MefistoSamael commented on VRisc/UseCases/VRisc.UseCases.csproj:
```diff
@@ -0,0 +1,30 @@
+    <ItemGroup>
+      <ProjectReference Include="..\..\Compiler\Data\Compiler.Data.csproj" />
+      <ProjectReference Include="..\Core\VRisc.Core.csproj" />
+      <ProjectReference Include="..\Infrastructure\VRisc.Infrastructure.csproj" />
```
use cases layer shouldn't has link to infrastructure

MefistoSamael commented on VRisc/Infrastructure/Interfaces/IEmulationStatesService.cs:
move to application

MefistoSamael commented on VRisc/Infrastructure/Interfaces/IEmulationTaskManager.cs:
move to applicattion (use cases)

uselessgoddess committed:
Merge pull request #13 from uselessgoddess/feat/add-tests

Add tests

uselessgoddess committed:
Merge branch 'feat/vrisc-service' into feat/api-gateway

uselessgoddess committed:
Merge pull request #14 from uselessgoddess/feat/api-gateway

Add Ocelot as api gateway

uselessgoddess committed:
Fix dependency rule between `UseCases` and `Infrastructure`

uselessgoddess committed:
Merge remote-tracking branch 'origin/dev' into feat/vrisc-service

# Conflicts:
#	Compiler/Core/Services/CompilerService.cs

uselessgoddess committed:
Bump server to be more client oriented
